File: backend-dotnet/Controllers/UsersController.cs
using backend_dotnet.Data.Dtos;
using backend_dotnet.Services;
using Microsoft.AspNetCore.Http.HttpResults;
using Microsoft.AspNetCore.Mvc;

namespace backend_dotnet.Controllers
{
    [Route("apiUsers")]
    [ApiController]
    public class UsersController : ControllerBase
    {
        private readonly UsersServices _usersService;

        public UsersController(UsersServices usersService)
        {
            _usersService = usersService;
        }

        [HttpGet("usersList")]
        public IActionResult UseresList([FromBody] UsersDto dto)
        {
            try
            {
                List<UsersDto> userList = _usersService.GetUsersList(dto);
                if (!userList.Any())
                {
                    return Conflict("Any users exists");
                }
                return Ok(userList);
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpGet("{userId}")]
        public IActionResult Get(string userId)
        {
            try
            {
                UsersDto? dto = _usersService.GetUser(userId);
                if (dto == null)
                {
                    return Conflict("Users doesn't exists");
                }
                return Ok(dto);
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpPost("createUsers")]
        public IActionResult CreateUsers([FromBody] UsersDto dto)
        {
            try
            {
                bool isCreated = _usersService.CreateUser(dto);
                if (isCreated) { return Ok("User created successfully"); } else { return Conflict("User ID or Mail is exists"); }
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }

        }

        [HttpPut("{userId}")]
        public IActionResult UpdateUser(string userId, [FromBody] UsersDto dto)
        {
            try
            {
                bool isUpdate = _usersService.UpdateUser(userId, dto);
                if (isUpdate) { return Ok("User Update successfully"); } else { return Conflict("Usernot exists or Mail is exists"); }
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpDelete("{userId}")]
        public IActionResult Delete(string userId)
        {
            try
            {
                bool isDeleted = _usersService.DeletedUsers(userId);
                if (isDeleted) { return Ok("Deleted is Successfully"); } else { return Conflict("Deleted is Successfully"); }
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }
    }
}

File: backend-python/controllers/users_controller.py
from fastapi import APIRouter, Depends, Body
from sqlalchemy.orm import Session
from database import get_db
from dtos.users_dto import UsersDto 
from services import users_service

controller = APIRouter(prefix="/apiUsers", tags=["Users"])

@controller.get("/userList")
def get_user_list(dto : UsersDto, db : Session = Depends(get_db)):
  try:
    result = users_service.get_user_list(dto, db)
    if not result:
      return "Users list is empty"
    return result
  except Exception as ex:
    return str(ex)

@controller.get("/{user_id}")
def get_user(user_id : str, db : Session = Depends(get_db)):
  try:
    result = users_service.get_user(user_id, db)
    if not result:
      return "User not exists"
    return result
  except Exception as ex:
    return str(ex)

@controller.post("/createUser")
def create_user(dto : UsersDto, db : Session = Depends(get_db)):
  try:
    result = users_service.create_user(dto, db)
    if result is False:
      return "User already exists or Id or mail is empty"
    return "Create User Successfully"
  except Exception as ex:
    return str(ex)

@controller.put("/{user_id}")
def update_user(user_id : str, dto : UsersDto, db : Session = Depends(get_db)):
  try:
    result = users_service.update_user(user_id, dto, db)
    if result is False:
      return "User not exists"
    if isinstance(result, str):
      return result
    return "Update User Successfully"
  except Exception as ex:
    return str(ex)

@controller.delete("/{user_id}")
def delete_user(user_id : str, db : Session = Depends(get_db)):
  try:
    result = users_service.delete_user(user_id, db)
    if result is False:
      return "User no exists"
    return "Delete User Successfully"
  except Exception as ex:
    return str(ex)